services/wallet/transfer/transaction_manager.go

```diff
@@ -26,13 +27,36 @@ type SignatureDetails struct {
 	V string `json:"v"`
 }
 
+func (sd *SignatureDetails) Validate() error {
+	if len(sd.R) != 64 || len(sd.S) != 64 || len(sd.V) != 2 {
+		return ErrInvalidSignatureDetails
+	}
+
+	return nil
+}
+
+// TODO: remove this struct once mobile switches to the new approach
 type TransactionDescription struct {
 	chainID   uint64
 	from      common.Address
 	builtTx   *ethTypes.Transaction
 	signature []byte
 }
 
+type RouterTransactionDetails struct {
+	routerPath         *routes.Path
+	txArgs             *transactions.SendTxArgs
+	tx                 *ethTypes.Transaction
+	txHashToSign       types.Hash
+	txSignature        []byte
+	txSentHash         types.Hash
+	approvalTxArgs     *transactions.SendTxArgs
+	approvalTx         *ethTypes.Transaction
+	approvalHashToSign types.Hash
+	approvalSignature  []byte
+	approvalTxSentHash types.Hash
```

Comment:
I'm not sure if it's used in API, but if it is, then please add explicit `json:"..."` tags

Reply:
No, it's not, it's just for internal use.